stl/inc/xutility

```diff
@@ -2005,6 +2004,21 @@ _NODISCARD constexpr size_t size(const _Ty (&)[_Size]) noexcept {
     return _Size;
 }
 
+#if _HAS_CXX20
+// FUNCTION TEMPLATE ssize
+template <class _Container>
+_NODISCARD constexpr auto ssize(const _Container& _Cont)
+    -> common_type_t<ptrdiff_t, make_signed_t<decltype(_Cont.size())>> {
+    using _Common = common_type_t<ptrdiff_t, make_signed_t<decltype(_Cont.size())>>;
+    return static_cast<_Common>(_Cont.size());
+}
+
+template <class _Ty, ptrdiff_t _Size>
+_NODISCARD constexpr ptrdiff_t ssize(const _Ty (&)[_Size]) noexcept {

```

Comment:
I want to see a bunch of testing for this overload. I suspect it will break intellisense for calls like ssize<long>(some_array) or ssize<long>({1,2,3,4})

Also, the fact that the array reference is not named may cause additional problems we don't know about.

(C array references seem to be a rarely used C++ feature where many frontend bugs lurk)

Reply:
That overload set is exactly what's in the paper being implemented here: http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2019/p1227r2.html

Reply:
I'm adding testing (which may or may not count as "a bunch of testing", but it's the same as what we have for `size()`). Note that `size()` has the same overload set, with the same unnamed reference-to-array: https://github.com/microsoft/STL/blob/1dcac20d280a52187da041fe0632b6b6373d5f1a/stl/inc/xutility#L1998-L2006

In my experience, while I've seen compiler bugs involving array rvalues, lvalue references to arrays don't appear to be unusually problematic, nor do unnamed references. The thing that most activates my compiler bug senses is the `common_type_t` involving a `decltype` which could have been a problem years ago, but appears to work robustly now.